client/app/bundles/course/assessment/submission/containers/VoiceResponseAnswer.jsx

```diff
@@ -0,0 +1,220 @@
+import StopIcon from 'material-ui/svg-icons/av/stop';
+import SingleFileInput from 'lib/components/redux-form/SingleFileInput';
+import recorderHelper from '../../utils/recorderHelper';
+import sharedConstants from '../../../../../lib/constants/sharedConstants';
+import {
+  setRecording,
+  setNotRecording,
+  recorderComponentMount,
+  recorderComponentUnmount,
+} from '../actions/index';
+
+const translations = defineMessages({
+  startRecord: {
+    id: 'course.assessment.submission.answer.startRecord',
+    defaultMessage: 'Start Record',
+  },
+  stopRecord: {
+    id: 'course.assessment.submission.answer.stopRecord',
+    defaultMessage: 'Stop Record',
+  },
+  chooseVoiceFileExplain: {
+    id: 'course.assessment.submission.answer.chooseVoiceFileExplain',
+    defaultMessage: 'Drag your voice file here, or click to select voice file. \
+                     Only wav and mp3 files are supported. Alternatively, you may press the\
+                     record button below to record through your microphone',
+  },
+});
+
+const styles = {
+  fileInputWrapper: {
+    width: '60%',
+  },
+  singleFileInputChildrenWrapper: {
+    width: '100%',
+    height: '100%',
+    display: 'table',
+  },
+  singleFileInputChildren: {
+    display: 'table-cell',
+    verticalAlign: 'middle',
+  },
+  errorStyle: {
+    color: red500,
+  },
+};
+
+class VoiceResponseAnswer extends Component {
+
+  static propTypes = {
+    answerId: PropTypes.number.isRequired,
+    readOnly: PropTypes.bool.isRequired,
+    question: PropTypes.shape({
+      id: PropTypes.number.isRequired,
+    }),
+    dispatch: PropTypes.func.isRequired,
+    recordingComponentId: PropTypes.string.isRequired,
+    recording: PropTypes.bool.isRequired,
+    intl: intlShape.isRequired,
+  }
+
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch(recorderComponentMount());
+  }
+
+  componentWillUnmount() {
+    const { dispatch } = this.props;
+    dispatch(recorderComponentUnmount());
+  }
+
+  onStartRecord = () => {
+    const { dispatch } = this.props;
+    const recordingComponentId = this.currentRecordingComponentId();
+    recorderHelper.startRecord().then(() => dispatch(setRecording({ recordingComponentId })));
+  }
+
+  onStopRecord = field => () => {
+    const { dispatch } = this.props;
+    recorderHelper.stopRecord().then((file) => {
+      const { input: { onChange, value = {} } } = field;
+      const { url, name } = value;
+      /**
+       * check SingleFileInput about the format of the single file
+       */
+      onChange({ file, url, name });
+      return dispatch((setNotRecording()));
+    });
+  }
+
+  /**
+   * It is just a unique Id for each component make use of recorder
+   */
+  currentRecordingComponentId = () => {
+    const { question } = this.props;
+    return `voice_response_${question.id}`;
+  }
+
+  isDropFileSupported = file => sharedConstants.SUPPORTED_VOICE_FILE_TYPES.some(fileType => file.type === fileType)
```

Comment:
Actually we can reuse a prop called `accept` in the react-dropzone implemented in the `SIngleFileInput`, rather than trying to implement your own check here.

Reply:
OK, I will use ```accept``` instead.